site/content/en/docs/Integration Patterns/high-density-gameservers.md

```diff
@@ -13,34 +13,104 @@ use of resources to run multiple concurrent game sessions from within a single `
 The tradeoff here is that this requires more management on behalf of the integrated game server process and external 
 systems, since it works around the common Kubernetes and/or Agones container lifecycle.
 
-Utilising the new allocation `gameServerState` filter as well as the existing ability to edit the 
+Here are two different approaches to solving this problem with Agones:
+
+## Session/Room Counters
+
+{{< beta title="Counters And Lists" gate="CountsAndLists" >}}
+
+Utilising the allocation `gameServerState` filter as well as the new ability to add Counter capacity and counts to
+`GameServer` records at both [allocation time]({{% ref "/docs/Reference/gameserverallocation.md" %}}), and from
+within the game server process, [via the SDK][sdk-counter], means Agones is able to atomically track how many sessions 
+are available on a given a `GameServer` from the list of potentially Ready or Allocated `GameServers` when making an
+allocation request.
+
+By also using Counters, we can provide Agones the allocation metadata it needs to pack appropriately across the high 
+density `GameServer` instances as well.
+
+<a href="../../../diagrams/high-density-counters.puml.png" target="_blank">
+<img src="../../../diagrams/high-density-counters.puml.png" alt="High Density Allocation Diagram (Session Counters)" />
+</a>
+
+### Example `GameServerAllocation`
+
+The below `Allocation` will first attempt to find a `GameServer` from the `Fleet` `simple-game-server` that is already
+Allocated and also available capacity under the `rooms` Counter.
+
+If an Allocated `GameServer` does not exist with available capacity, then use the next selector to allocate a Ready
+`GameServer` from the `simple-game-server` `Fleet`.
+
+Whichever condition is met, once allocation is made against a `GameServer`, the `rooms` Counter will be incremented by
+one, thereby decrementing the available capacity of the `room` Counter on the `GameServer` instance. Once there is no 
+available capacity on the most full `GameServer`, the allocation will look for the next least full `GameServer` to
```

Comment:
If you're packing game servers, would you want to use `Descending` to pack the most full rooms?

Reply:
it's ascending because it's by available capacity. I remember us going back and forth on this - but it means that sorting on Allocation and Fleets is the same value for packing appropriately.

Reply:
Oh yep, that's right. The comment in line 59 threw me off.